addons/mail/static/src/discuss/discuss.js

```diff
@@ -31,6 +41,42 @@ export class Discuss extends Component {
     startCall() {
         this.messaging.startCall(this.messaging.discuss.threadId);
     }
+
+    /**
+     * @param {KeyboardEvent} ev
+     */
+    onKeyDownThreadNameInput(ev) {
+        switch (ev.key) {
+            case 'Enter':
+                this.renameThread();
+                this.threadNameInputRef.el.blur();
+                break;
+            case 'Escape':
+                this.state.threadRename = this.currentThread().name;
+                this.threadNameInputRef.el.blur();
+                break;
+        }
+    }
+
+    async renameThread() {
+        const newName = this.state.threadRename.trim();
+        const thread = this.currentThread();
+        if (
+            (newName && newName !== thread.name && thread.type === 'channel') ||
+            (newName !== thread.name && (thread.type === 'chat' || thread.type === 'group'))
+        ) {
+            await this.messaging.notifyThreadNameToServer(thread.id, newName);
+        }
+    }
+
+    /**
+     * @param {MouseEvent} ev
+     */
+    _onClickCaptureGlobal(ev) {
+        if (!this.threadNameInputRef.el.contains(ev.target)) {
+            this.renameThread();
+        }
+    }
```

Comment:
This function looks unnecessary. Can just use arrow function in `useExternalListener`.